pkg/csi_driver/controller_test.go

```diff
@@ -1252,76 +1269,87 @@ func ValidateExpectedError(t *testing.T, errResp <-chan error, operationUnblocke
 }
 
 func TestCreateSnapshot(t *testing.T) {
-	type BackupInfo struct {
-		s              *file.ServiceInstance
-		backupName     string
-		backupLocation string
+	type BackupTestInfo struct {
+		backup *file.BackupInfo
+		state  string
 	}
 	backupName := "mybackup"
 	project := "test-project"
 	zone := "us-central1-c"
 	region := "us-central1"
 	instanceName := "myinstance"
 	shareName := "myshare"
+	defaultBackupUri := fmt.Sprintf("projects/%s/locations/%s/backups/%s", project, region, backupName)
 	cases := []struct {
 		name          string
 		req           *csi.CreateSnapshotRequest
-		initialBackup *BackupInfo
+		resp          *csi.CreateSnapshotResponse
+		initialBackup *BackupTestInfo
 		expectErr     bool
 	}{
 		// Failure test cases
 		{
-			name: "Create snapshot request for multishare backed volumes",
+			name: "Existing backup found, with different volume Id (source zonal filestore instance), error expected",
 			req: &csi.CreateSnapshotRequest{
-				SourceVolumeId: "modeMultishare/mysc/test-project/location/myinstance/myshare",
+				SourceVolumeId: modeInstance + "/" + zone + "/" + "myinstance1" + "/" + shareName,
+				Name:           backupName,
+				Parameters: map[string]string{
+					util.VolumeSnapshotTypeKey: "backup",
+				},
+			},
+			initialBackup: &BackupTestInfo{
+				backup: &file.BackupInfo{
+					Project:            project,
+					Location:           region,
+					SourceInstanceName: instanceName,
+					SourceShare:        shareName,
+					Name:               backupName,
+					SourceVolumeId:     modeInstance + "/" + zone + "/" + instanceName + "/" + shareName,
+				},
 			},
 			expectErr: true,
 		},
 		{
-			name: "Existing backup found, with different volume Id (source zonal filestore instance), error expected",
+			name: "Existing backup found in state CREATING",
 			req: &csi.CreateSnapshotRequest{
-				SourceVolumeId: "modeInstance/us-central1-c/myinstance1/myshare",
+				SourceVolumeId: "modeInstance/us-central1/myinstance/myshare",
 				Name:           backupName,
 				Parameters: map[string]string{
 					util.VolumeSnapshotTypeKey: "backup",
 				},
 			},
-			initialBackup: &BackupInfo{
-				s: &file.ServiceInstance{
-					Project:  project,
-					Location: zone,
```

Comment:
While the initial service instance can be a zone, the initialBackup itself cannot be. Having the service instance in the backupInfo made the tests more difficult to parse through.